src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -493,16 +497,16 @@ private void handleMalformedDocResponse(BulkItemResponse response) {
       case FAIL:
       default:
         log.error(
-            "Encountered an illegal document error '{}'. To ignore future records like this,"
-                + " change the configuration '{}' to '{}'.",
-            response.getFailureMessage(),
+            "Encountered an illegal document error -> Response status: '{}',\n "
+                    + "Index: '{}',\n Document Id: '{}'\n"
+                    + " To ignore future records like this,"
+                    + " change the configuration '{}' to '{}'.",
+            response.getFailure().getStatus(),
+            response.getFailure().getIndex(),
+            response.getFailure().getId(),
             ElasticsearchSinkConnectorConfig.BEHAVIOR_ON_MALFORMED_DOCS_CONFIG,
             BehaviorOnMalformedDoc.IGNORE
         );
-        error.compareAndSet(
```

Comment:
What is the impact of removing this statement?
Do we not need it?

Reply:
Thanks Subhash. After careful consideration again, I think we should keep this. I missed the fact that this error might be getting used somewhere else in the code. It is necessary to set the connect exception properly. I have made the necessary changes. Please review.